R/subscript-tables.R

```diff
@@ -607,11 +609,12 @@ updateQuestionTypesFromArgs <- function(dropped.dims, question.type) {
     question.type
 }
 
-getFallbackQuestionType <- function(question.types) {
+getFallbackQuestionType <- function(question.types, all.relevant.dims.dropped) {
     n.question.types <- length(question.types)
-    if (n.question.types == 1L) {
-        if (question.types == "NumberMulti") return("Number")
-        return(question.types)
+    if (n.question.types == 1L || all.relevant.dims.dropped) {
+        if (question.types[1L] %in% c("NumberMulti", "NumberGrid")) return("Number")
+        else if (question.types[1L] %in% c("PickOneMulti", "PickAnyGrid")) return("PickAny")
```

Comment:
`else if` still redundant with the `return` used in the `if` block

I believe `all.relevant.dim.dropped` will always be `TRUE` when this function is called. So it will always enter the `if` statement starting on the new line 614 so the code after line 618 never runs.

When there are two question types, it picks the first one to decide how to give a final question type. If the questions types are `c("Date", "NumberGrid")` then this will produce an output type of `Date` which is incorrect.

I think the `all.relevant.dims.dropped` should be removed from this function and a more comprehensive function made. I'll make a draft.